Lib/test/test_pyrepl/test_interact.py

```diff
@@ -119,13 +119,38 @@ def test_runsource_shows_syntax_error_for_failed_compilation(self):
 
     def test_no_active_future(self):
         console = InteractiveColoredConsole()
-        source = "x: int = 1; print(__annotate__(1))"
+        source = dedent("""\
+        x: int = 1
+        print(__annotate__(1))
+        """)
         f = io.StringIO()
         with contextlib.redirect_stdout(f):
             result = console.runsource(source)
         self.assertFalse(result)
         self.assertEqual(f.getvalue(), "{'x': <class 'int'>}\n")
 
+    def test_future_annotations(self):
+        console = InteractiveColoredConsole()
+        source = dedent("""\
+        from __future__ import annotations
+        def g(x: int): ...
+        print(g.__annotations__)
+        """)
+        f = io.StringIO()
+        with contextlib.redirect_stdout(f):
+            result = console.runsource(source)
+        self.assertFalse(result)
+        self.assertEqual(f.getvalue(), "{'x': 'int'}\n")
+
+    def test_future_barry_as_flufl(self):
+        console = InteractiveColoredConsole()
+        f = io.StringIO()
+        with contextlib.redirect_stdout(f):
+            result = console.runsource("from __future__ import barry_as_FLUFL\n")
+            result = console.runsource("""print("black" <> 'blue')\n""")
```

Comment:
This is required because the parser can't deal with the future import changing the flags inside a run. This is true now:

```python
❯ ./python.exe -c "from __future__ import barry_as_FLUFL; 1<>0"
  File "<string>", line 1
    from __future__ import barry_as_FLUFL; 1<>0
                                            ^^
SyntaxError: invalid syntax
```

but was also true in the old parser:
```python
❯ python3.9 -Xoldparser -c "from __future__ import barry_as_FLUFL; 1<>0"
  File "<string>", line 1
    from __future__ import barry_as_FLUFL; 1<>0
                                             ^
SyntaxError: invalid syntax
```

The consequence of that with PyREPL is that you cannot paste multiline blocks with that future and make them run, because PyREPL executes everything save for the last line in one go.

PyREPL isn't doing anything wrong here, it's just exposing the bug that was always there. Therefore, I'm not sure we should be working around this pasting thing for this one future. Other futures work fine, as you can see in the test added above.